### PR TITLE
Update AzureCI matrix for Python 3.11/NumPy 1.21..1.24

### LINKED ISSUE #8834: Clean up AzureCI matrix
> The CI matrix will probably need some further adjustment, as will general build configuration, but not for this PR as it seems out of scope.

_Originally posted by @stuartarchibald in https://github.com/numba/numba/issues/8822#issuecomment-1478350653_

#8822 PR may have increased the number of CI configuration too much. We should consider reducing it and perhaps remove testing of older python versions.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This updates the AzureCI test matrix. It aims to ensure that:

* The latest supported NumPy and Python are tested on all OS.
* The oldest supported NumPy and Python are tested on all OS.
* That the rest of the "matrix" formed from the outer product of supported NumPys x supported Pythons has at least one build if packages are available.

Closes #8834

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Close/Open to trigger new CI build against fixed `main`.

**Comment by sklam** [comment]
CI is failing because there is no scipy for numpy=1.22 and python=3.11

**Comment by sklam** [comment]
The window llvm error is tracked at https://github.com/numba/numba/issues/8840

**Comment by stuartarchibald** [comment]
> CI is failing because there is no scipy for numpy=1.22 and python=3.11

Thanks, 5424212 fixes.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -17,17 +17,11 @@ jobs:
         NUMPY: '1.21'
         CONDA_ENV: 'azure_ci'
         TEST_START_INDEX: 0
-      py310_np124:
-        PYTHON: '3.10'
-        NUMPY: '1.24'
-        CONDA_ENV: 'azure_ci'
-        TEST_THREADING: 'tbb'
```
Comment (sklam) [inline azure-pipelines.yml:24]: This remove TBB testing on OSX

Comment (stuartarchibald) [reply azure-pipelines.yml:24]: Should probably keep on doing this as it tests a different loading path. Reinstated in 1bd9dd0.

**Pull request merged**